### DISCUSSION TITLE
How do I download subtitles?

### DISCUSSION THREAD

**Discussion by guzz46** [tomasklaen/uosc#923] [Q&A] [answered]
Clicking on the subtitles icon brings up a list of of subtitles which is great, but if I want to search online for different subtitles when I click on the download option and then click the search icon I get a 'process exited with code -3' error, when run from the terminal it just displays [uosc]

Thanks

**Comment by christoph-heinrich** [comment]
Are the ziggy binary(s) in the correct location? I get that error when it can't find the binary.

It should be in `~~/scripts/uosc/bin/` with a filename like `ziggy-windows.exe`, `ziggy-darwin` or `ziggy-linux`.

**Reply by guzz46** [reply]
I have ~/.config/mpv/scripts/uosc/bin/ziggy-linux

**Reply by christoph-heinrich** [reply]
Maybe a lack of permissions? Make sure it's marked as executable.

**Reply by guzz46** [reply]
Hmm, its marked as executable, but I'm running nixos, and usually if I want to run a typical script I need to put bash in front of it, so 'bash ~/script'
I wonder if I can edit a file so instead of calling ~/.config/mpv/scripts/uosc/bin/ziggy-linux, it calls bash ~/.config/mpv/scripts/uosc/bin/ziggy-linux?

**Comment by guzz46** [comment, marked as answer]
I solved it, I think the problem was due to having my /home directory mounted as noexec, so the ziggy file wouldn't execute when called, so I copied the ziggy file to my scripts folder which is mounted as exec, and then I edited the main.lua file to point to the new location of the ziggy file, so it now looks like this

```
do
	local bin = 'ziggy-' .. (state.platform == 'windows' and 'windows.exe' or state.platform)
	config.ziggy_path = '~/.config/scripts/ziggy-linux'
end

**Reply by christoph-heinrich** [reply]
You could try exporting an environment variable `MPV_UOSC_ZIGGY` so you don't have to edit the lua file, otherwise you'll have to edit it every time you update uosc.

**Reply by guzz46** [reply]
I might look into that, thanks

Edit: I ended up restoring the default main.lua and putting the following in my ~/.profile

``
export MPV_UOSC_ZIGGY=~/.config/scripts/ziggy-linux
``
